py2puml/cli.py

```diff
@@ -1,29 +1,36 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-import argparse
+from argparse import ArgumentParser
+from sys import argv
 
 from py2puml.py2puml import py2puml
-from py2puml import __version__
+from py2puml import __version__, __description__
 
+CLI_VERSION = f'py2puml {__version__}'
+PATH_ARG_HELP = 'the path of the domain'
+MODULE_ARG_HELP = 'the module of the domain'
 
+# CLI end-point when py2puml is installed at the system level
 def run():
-    argparser = argparse.ArgumentParser(description='Generate PlantUML diagrams to document your python code.')
+    argparser = ArgumentParser(description=__description__)
 
-    argparser.add_argument('-v', '--version', action='version',
-                           version='py2puml {version}'.format(version=__version__))
-
-    argparser.add_argument('path', metavar='path',
-                           type=str,
-                           help='the path of the domain.')
-
-    argparser.add_argument('module', metavar='module',
-                           type=str,
-                           help='the module of the domain.',
-                           default=None)
+    argparser.add_argument('-v', '--version', action='version', version=CLI_VERSION)
+    argparser.add_argument('path', metavar='path', type=str, help=PATH_ARG_HELP)
+    argparser.add_argument('module', metavar='module', type=str, help=MODULE_ARG_HELP, default=None)
 
     args = argparser.parse_args()
-
-    print(''.join(
-        py2puml(args.path, args.module)
-    ))
+    print(''.join(py2puml(args.path, args.module)))
+
+# CLI end-point when py2puml is NOT installed at the system level
+if __name__ == '__main__':
+    if len(argv) == 2 and argv[1] in ['-v', '--version']:
+        print(f'{__description__} ({CLI_VERSION})')
+    elif len(argv) == 3:
+        print(''.join(py2puml(argv[1], argv[2])))
+    else:
+        raise ValueError(f'''
+The CLI can be called in 2 ways:
+- version: python -m py2puml.cli -v or python -m py2puml.cli --version
+- plantuml documentation (2 mandatory arguments): `python py2puml.cli domain/path domain.module`: {PATH_ARG_HELP}, {MODULE_ARG_HELP}
+''')
```

Comment:
To me this is kind of convoluted and `argparse` usually takes care of CLI mis-usage, be it installed on the system or not.  🙂

It might be possible to make the CLI accessible with a similar and conciser syntax, for instance :
```sh
$ python -m py2puml
```

by modifying `py2pulm.__init__` to include:

```python
from py2puml.cli import run

if __name__ == "__main__":
    run()
```

This is to test though.

Reply:
yes, I didn't feel right implementing the CLI twice with `sys.argv`. Can you try modifying `py2pulm.__init__` as you suggest and remove the convoluted code, please?

Reply:
I'll do it.